src/compile/axis.ts

```diff
@@ -137,7 +138,7 @@ export function title(model: Model, channel: Channel) {
   }
 
   // if not defined, automatically determine axis title from field def
-  const fieldTitle = model.fieldTitle(channel);
+  const fieldTitle = fieldDefTitle(model.fieldDef(channel), axis.showUnit === 'title' || !axis.showUnit);
```

Comment:
showUnit is always a string, and the default value is `'title'`. 

I think you don't need `!axis.showUnit` here.